CondTools/SiPhase2Tracker/plugins/DTCCablingMapProducer.cc

```diff
@@ -178,30 +195,41 @@ void DTCCablingMapProducer::LoadModulesToDTCCablingMapFromCSV(
           uint32_t detIdRaw;
 
           try {
-            detIdRaw = std::stoi(csvColumn[csvFormat_idetid_]);
+            detIdRaw = std::stoi(csvColumn.at(csvFormat_idetid_));
           } catch (std::exception const& e) {
             if (verbosity_ >= 0) {
               edm::LogError("CSVParser") << "-- malformed DetId string in CSV file: \"" << csvLine << "\"" << endl;
             }
             throw e;
           }
 
-          unsigned const dtc_id = strtoul(csvColumn[csvFormat_idtcid_].c_str(), nullptr, 10);
+          unsigned const dtc_id = strtoul(csvColumn.at(csvFormat_idtcid_).c_str(), nullptr, 10);
           unsigned gbt_id;
           unsigned elink_id;
 
-          if (generate_fake_valid_gbtlink_and_elinkid_) {
-            for (gbt_id = gbt_id_minvalue; gbt_id < gbt_id_maxvalue + 1u; ++gbt_id) {
+          switch (dummy_fill_mode_) {
+            default:
+            case DUMMY_FILL_DISABLED:
+              gbt_id = strtoul(csvColumn.at(csvFormat_igbtlinkid_).c_str(), nullptr, 10);
+              elink_id = strtoul(csvColumn.at(csvFormat_ielinkid_).c_str(), nullptr, 10);
+              break;
+            case DUMMY_FILL_ELINK_ID:
+              gbt_id = strtoul(csvColumn.at(csvFormat_igbtlinkid_).c_str(), nullptr, 10);
               for (elink_id = elink_id_minvalue; elink_id < elink_id_maxvalue + 1u; ++elink_id) {
                 if (!(pCablingMap_->knowsDTCELinkId(DTCELinkId(dtc_id, gbt_id, elink_id))))
-                  goto gbtlink_and_elinkid_generator_end;  //break out of this double loop
+                  break;
               }
-            }
-          gbtlink_and_elinkid_generator_end:
-            ((void)0);  // This is a NOP, it's here just to have a valid (although dummy) instruction after the goto tag
-          } else {
-            gbt_id = strtoul(csvColumn[csvFormat_igbtlinkid_].c_str(), nullptr, 10);
-            elink_id = strtoul(csvColumn[csvFormat_ielinkid_].c_str(), nullptr, 10);
+              break;
+            case DUMMY_FILL_ELINK_ID_AND_GBT_ID:
+              for (gbt_id = gbt_id_minvalue; gbt_id < gbt_id_maxvalue + 1u; ++gbt_id) {
```

Comment:
Same here, but with the GBT id. There are at most 72 GBTs (6 trunk cables x 12 GBTS) connected to 1 DTC.
gbt_id_minvalue = 0;
gbt_id_maxvalue = 72;
So why not 
(gbt_id = gbt_id_minvalue; gbt_id < gbt_id_maxvalue; ++gbt_id)

Reply:
Good, you caught it. I will change it!